src/databases/Blueprint/avtBlueprintFileFormat.C

```diff
@@ -1593,6 +1812,166 @@ avtBlueprintFileFormat::GetVectorVar(int domain, const char *varname)
+        df = avtMaterial::Destruct;
+    }
+
+    return rv;
+}
+
+
+// ****************************************************************************
+//  Method: avtBlueprintFileFormat::GetMaterial
+//
+//  Purpose:
+//      Gets the auxiliary data from a Blueprint Database.
+//
+//  Arguments:
+//      domain     The domain of interest.
+//      mat_name   The material of interest.
+//
+//  Returns:    The auxiliary data.  Throws an exception if this is not a
+//              supported data type.
+//
+//  Programmer: Cyrus Harrison
+//  Creation:   December 8, 2020
+//
+//  Modifications:
+//
+//
+// ****************************************************************************
+
+avtMaterial *
+avtBlueprintFileFormat::GetMaterial(int domain,
+                                    const char *mat_name)
+{
+    BP_PLUGIN_INFO("avtBlueprintFileFormat::GetMaterial " 
+                    << domain << " "
+                    << mat_name);
+
+    Node n_matset;
+    ReadBlueprintMatset(domain,
+                        mat_name,
+                        n_matset);
+
+    std::vector<std::string> matnames = n_matset["matnames"].child_names();
+
+    // use to_silo util to convert from bp to the mixslot rep
+    // that silo and visit use
+
+    Node n_silo_matset;
+    conduit::blueprint::mesh::matset::to_silo(n_matset,
+                                              n_silo_matset);
+
+    int nmats = (int) matnames.size();
+    int nzones = (int) n_silo_matset["matlist"].dtype().number_of_elements();
+
+    int *matlist  = NULL;
+    int *mix_mat  = NULL;
+    int *mix_next = NULL;
+
+    // we need int ptrs for the avtMaterial object,
+    // convert if needed
+
+    Node n_tmp;
+    if(!n_silo_matset["matlist"].dtype().is_int())
+    {
+        n_silo_matset["matlist"].to_int_array(n_tmp["matlist"]);
+        n_silo_matset["mix_mat"].to_int_array(n_tmp["mix_mat"]);
+        n_silo_matset["mix_next"].to_int_array(n_tmp["mix_next"]);
+        matlist  = n_tmp["matlist"].as_int_ptr();
+        mix_mat  = n_tmp["mix_mat"].as_int_ptr();
+        mix_next = n_tmp["mix_next"].as_int_ptr();
+    }
+    else
+    {
+        matlist  = n_silo_matset["matlist"].as_int_ptr();
+        mix_mat  = n_silo_matset["mix_mat"].as_int_ptr();
+        mix_next = n_silo_matset["mix_next"].as_int_ptr();
+    }
+
+    // we need to adjust the matlist.
+    for(int i=0;i<nzones;i++)
+    {
+        if(matlist[i] > 0 )
+        {
+            matlist[i]--;
+        }
+    }
+
+    int mix_len  = (int) n_silo_matset["mix_mat"].dtype().number_of_elements();
+
+    float *mix_vf = NULL;
+    if(n_silo_matset["mix_vf"].dtype().is_float())
+    {
+        mix_vf = n_silo_matset["mix_vf"].as_float_ptr();
+    }
+    else
+    {
+        n_silo_matset["mix_vf"].to_float_array(n_silo_matset["mix_vf_float"]);
+        mix_vf = n_silo_matset["mix_vf_float"].as_float_ptr();
+    }
+
+    // we are using the following avtMaterial constructor:
```

Comment:
Does this comment add any value?

Reply:
this was helpful when doing the work, but I'll remove this now -- given that the call now has extra comments that help.